src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPConnectorApp.scala

```diff
@@ -28,47 +28,81 @@ object ReadingSFTPConnectorApp extends IOApp {
           config.spark.serializer.contains("KryoSerializer").toString
         )
         .set("spark.hadoop.fs.sftp.impl", "org.apache.hadoop.fs.sftp.SFTPFileSystem")
+        .set("spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation","true")
         .registerKryoClasses(RegisterInKryo.classes.toArray)
 
       sparkSession: SparkSession = SparkSession.builder
         .config(defaultSparkConf)
         .enableHiveSupport
         .getOrCreate()
 
+      sftpUser1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_USER")
+        .getOrElse(config.sftp.sftpUser)
+      sftpPass1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PASS")
+        .getOrElse(config.sftp.sftpPass)
+      sftpHost1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_HOST")
+        .getOrElse(config.sftp.sftpHost)
+      sftpPath1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PATH")
+        .getOrElse(config.sftp.sftpPath)
+
+      _ = println(s"#### From SparkConf: $sftpUser1, $sftpHost1, $sftpPath1")
+
       // Construct Spark data frame reading a file from SFTP
       data: DataFrame = sparkSession.read
         .format("com.springml.spark.sftp") //TODO: Is this library single threaded?
-        .option("host", config.sftp.sftpHost)
-        .option("username", config.sftp.sftpUser)
-        .option("password", config.sftp.sftpPass)
+        .option("host", sftpHost1)
+        .option("username", sftpUser1)
+        .option("password", sftpPass1)
+        .option("header", true)
         .option("fileType", "csv")
+        .option("delimiter", ",")
         .option("inferSchema", "true")
-        .load("/tmp/spark/sample.psv")
-        .repartition(config.spark.partitions)
-
-      // Some other sample operations
-      //_ = sparkSession.catalog.listDatabases().show(truncate = false)
-      //_ = sparkSession.catalog.listTables().show(truncate = false)
-      //_ = sparkSession.sql("show tables").show(truncate = false)
+        .load(sftpPath1)
 
+      //Testing the content of the dataframe, the time in doing the count can be using to measure time in reading.
       _ = data.printSchema()
-      _ = println(s"##############COUNT: ${data.count()}")
+      _ = println(s"### COUNT: ${data.count()}")
       _ = data.show(false)
 
-      // TODO: Other steps to do after the processing.
+      // Creating databases do not work in Dataproc: https://github.com/mozafari/verdictdb/issues/163
```

Comment:
So this means we have not explored to put the CSV information in HIVE tables?

Reply:
It is persisted in Hive, we can't not use other databases other than the default in Dataproc.